tensorbay/opendataset/nuScenes/loader.py

```diff
@@ -0,0 +1,279 @@
+                    instance.json
+                    log.json
+                    map.json
+                    sample_annotation.json
+                    sample_data.json
+                    sample.json
+                    scene.json
+                    sensor.json
+                    visibility.json
+            v1.0-test/
+                maps/
+                samples/
+                sweeps/
+                v1.0-test/
+            v1.0-trainval/
+                maps/
+                samples/
+                sweeps/
+                v1.0-trainval/
+
+    Arguments:
+        path: The root directory of the dataset.
+
+    Returns:
+        Loaded :class:`~tensorbay.dataset.dataset.FusionDataset` instance.
+
+    """
+    root_path = os.path.abspath(os.path.expanduser(path))
+
+    dataset = FusionDataset(DATASET_NAME)
+    dataset.notes.is_continuous = True
+    dataset.notes.bin_point_cloud_fields = ["X", "Y", "Z", "Intensity", "Ring"]
+    dataset.load_catalog(os.path.join(os.path.dirname(__file__), "catalog.json"))
+
+    for subset in os.listdir(root_path):
+        subset_path = os.path.join(root_path, subset)
+        for segment in _generate_segments(subset_path, subset):
+            dataset.add_segment(segment)
+
+    return dataset
+
+
+def _generate_segments(subset_path: str, subset: str) -> Iterator[FusionSegment]:
+    annotation_files = _load_annotation_files(os.path.join(subset_path, subset))
+
+    for scene in annotation_files["scenes"]:
+        segment = FusionSegment(f"{subset}/{scene['name']}")
+        segment.description = scene["description"]
+        current_frame = scene["first_sample_token"]
+        while current_frame:
+            frame, lidar_ego_pose_token = _load_frame(
+                current_frame,
+                annotation_files,
+                segment.sensors,
+                subset_path,
+            )
+            if "test" not in subset:
+                frame["LIDAR_TOP"].label.box3d = _load_labels(
+                    current_frame,
+                    segment.sensors["LIDAR_TOP"].extrinsics,
+                    lidar_ego_pose_token,
+                    annotation_files,
+                )
+
+            segment.append(frame)
+            current_frame = annotation_files["samples"][current_frame]["next"]
+        yield segment
+
+
+def _load_annotation_files(info_path: str) -> Dict[str, Any]:
+    annotation_files = {
+        "samples": _load_info(os.path.join(info_path, "sample.json")),
+        "frame_data": _load_sample_info(os.path.join(info_path, "sample_data.json")),
+        "calibrated_sensors": _load_info(os.path.join(info_path, "calibrated_sensor.json")),
+        "ego_poses": _load_info(os.path.join(info_path, "ego_pose.json")),
+        "sensor": _load_info(os.path.join(info_path, "sensor.json")),
+    }
+    with open(os.path.join(info_path, "scene.json"), "r") as file:
+        annotation_files["scenes"] = json.load(file)
+    if "test" not in info_path:
```

Comment:
`info_path` is a abspath, what if nuScene dataset files is put under a directory named "test"?